syntax = "proto3";

package grpcgateway;

import "annotations.proto";

// Services

service PredictionService {
  rpc Predict(PredictRequest) returns (PredictResponse) {
    option (google.api.http) = {
      post: "/predict"
      body: "*"
    };
  }
}

service DataService {
  rpc GetData(GetDataRequest) returns (GetDataResponse) {
    option (google.api.http) = {
      get: "/data/{id}"
    };
  }
}

service AccountService {
  rpc GetUser(GetUserRequest) returns (GetUserResponse) {
    option (google.api.http) = {
      get: "/user/{id}"
    };
  }
}

// Messages

message PredictRequest {
  // Request fields for prediction
}

message PredictResponse {
  // Response fields for prediction
}

message GetDataRequest {
  string id = 1;
}

message GetDataResponse {
  // Response fields for data
}

message GetUserRequest {
  string id = 1;
}

message GetUserResponse {
  // Response fields for user account
}
